test/action.yml

```diff
@@ -0,0 +1,82 @@
+---
+name: Test Nodejs
+description: |
+  Github Action to Build & Test Nodejs
+
+  RequiredEnv:
+    GITHUB_TOKEN
+    SONAR_TOKEN
+inputs:
+  npm_test_script_name:
+    description: npm test script name
+    required: false
+    default: test
+  nodejs-version:
+    description: >
+      The nodejs-version input is optional.
+      The default version of Nodejs in PATH varies
+      between runners and can be changed unexpectedly
+      so we recommend always setting Nodejs version explicitly
+      using the nodejs-version input.
+    required: false
+runs:
+  using: composite
+  steps:
+    - name: Find Nodejs Packager
+      id: packager
+      shell: bash
+      run: |
+        echo "packager=yarn" >> $GITHUB_OUTPUT
+        if [ -f "package-lock.json" ]; then
+            echo "packager=npm" >> $GITHUB_OUTPUT
+        fi
+
+    - name: Set up Nodejs
+      uses: actions/setup-node@v3
+      with:
+        node-version: ${{ inputs.nodejs-version }}
+        cache: ${{ steps.packager.outputs.packager }}
+
+    - name: Run pre_test.sh
+      shell: bash
+      run: |
+        pretest_cmd="echo 'NO_PRETEST'"
+        if [ -f ".github/actions/pre_test.sh" ]; then
+            pretest_cmd="$(cat .github/actions/pre_test.sh)"
+        fi
+        echo "pretest_cmd: $pretest_cmd"
+        sh -c "${pretest_cmd}"
+
+    - name: Install Application node_modules
+      shell: bash
+      run: |
+        echo "::group::Upgrade Packages"
+        if [ -f "yarn.lock" ]; then
+            yarn install --frozen-lockfile
+        fi
+        if [ -f "package-lock.json" ]; then
+            npm install
+            npm ci
+        fi
+        echo "::endgroup::"
+
+    - name: Run Test
+      shell: bash
+      run: |
+        echo "::group::Run Test"
+        if [ -f "yarn.lock" ]; then
+            yarn run ${{ inputs.npm_test_script_name }}
+        fi
+        if [ -f "package-lock.json" ]; then
+           npm run ${{ inputs.npm_test_script_name }}
+        fi
+        echo "::endgroup::"
+
+    - name: SonarCloud Scan
+      uses: sonarsource/sonarcloud-github-action@master
+      with:
+        args: >
+          -Dsonar.qualitygate.wait=${{ steps.sonar-setup.outputs.wait_flag }}
+          -Dsonar.projectKey=${{ env.SONAR_PROJECT_KEY }}
+          -Dsonar.organization=${{ env.SONAR_ORG }}
+          -Dsonar.javascript.lcov.reportPaths=coverage/lcov.info
```

Comment:
Can we add a note to document that the coveragePath should be here, or can it be added as a custom variable?